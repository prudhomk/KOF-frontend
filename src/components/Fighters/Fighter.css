.Fighter a {
  list-style: none;
  background: white;
  border: 1px solid darkgray;
  padding: 10px;

  display: grid;
  grid-template-rows: 30px 40px 120px 20px;
  grid-gap: 10px;

  justify-items: center;

  text-decoration: none;
  color: inherit;
  cursor: pointer;
}

.Fighter h2 {
  margin: 0;
}

.Fighter img {
  height: 100%;
  width: 100%;
  object-fit: contain;
}

.Fighter p {
  margin: 0;
  font-size: .9rem;
}