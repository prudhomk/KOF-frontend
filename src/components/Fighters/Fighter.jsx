import React from 'react';
import PropTypes from 'prop-types';
// import './Fighter.css';

const Fighter = ({ name, birthplace, style, image }) => (
  <figure className="Fighter">
    <h2>Name: {name}</h2>
    <img src={image} alt={name} />
    <figcaption>
      <p>Birthplace: {birthplace}</p>
      <p>Fighting Style: {style}</p>
    </figcaption>
   
  </figure>
);

Fighter.propTypes = {
  name: PropTypes.string.isRequired,
  birthplace: PropTypes.string.isRequired,
  style: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};

export default Fighter;
