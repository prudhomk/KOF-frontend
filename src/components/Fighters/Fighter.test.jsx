import React from 'react';
import { render } from '@testing-library/react';
import Fighter from './Fighter';
import { MemoryRouter } from 'react-router-dom';

describe('Villager component', () => {
  it('renders Villager', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Fighter
          name="Bali"
          birthplace="Egypt"
          style="kendo"
          image="https://image.com/image.jpeg"
        />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
