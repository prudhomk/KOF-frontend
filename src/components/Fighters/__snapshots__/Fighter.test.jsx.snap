// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Villager component renders Villager 1`] = `
<DocumentFragment>
  <figure
    class="Fighter"
  >
    <h2>
      Name: Bali
    </h2>
    <img
      alt="Bali"
      src="https://image.com/image.jpeg"
    />
    <figcaption>
      <p>
        Birthplace: Egypt
      </p>
      <p>
        Fighting Style: kendo
      </p>
    </figcaption>
  </figure>
</DocumentFragment>
`;
